src/Microsoft.AspNet.Mvc.Razor/Compilation/RoslynCompilationService.cs

```diff
@@ -89,6 +94,8 @@ public CompilationResult Compile(RelativeFileInfo fileInfo, string compilationCo
                 throw new ArgumentNullException(nameof(compilationContent));
             }
 
+            _logger.GeneratedCodeToAssemblyCompilationStart(fileInfo.RelativePath);
```

Comment:
Look at https://github.com/aspnet/Mvc/commit/c125b4e59badf551c85bbc8d6ae8cb133fe6de7c for the timing pattern.